capa/features/extractors/cape/models.py

```diff
@@ -0,0 +1,456 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at: [package root]/LICENSE.txt
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+#  is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+import binascii
+from typing import Any, Dict, List, Tuple, Union, Optional
+
+from pydantic import Field, BaseModel, ConfigDict
+from typing_extensions import Annotated, TypeAlias
+from pydantic.functional_validators import BeforeValidator
+
+
+def validate_hex_int(value):
+    return int(value, 16) if isinstance(value, str) else value
+
+
+def validate_hex_bytes(value):
+    return binascii.unhexlify(value) if isinstance(value, str) else value
+
+
+HexInt = Annotated[int, BeforeValidator(validate_hex_int)]
+HexBytes = Annotated[bytes, BeforeValidator(validate_hex_bytes)]
+
+
+class Model(BaseModel):
+    model_config = ConfigDict(extra="forbid")
```

Comment:
we can turn this off, which will:

  1. no longer raise an exception when strictly new, unexpected data is seen, but
  2. at the expense that we can't as quickly find updates to the model

maybe we want to turn this off during releases.

https://docs.pydantic.dev/latest/usage/models/#extra-fields